### ISSUE TITLE
Service metrics support via Generic Extensions

### ISSUE THREAD

**Issue by gberche-orange** [openservicebrokerapi/servicebroker#485]
As a service user:
* in order to use services managed by a 3rd party provider but still: 
   * monitor my app usage of the service. E.g. 
      * getting close to quotas/define service plans, and I need to update plans, 
      * suboptimal service usage by my app that requires optimization from app developers
   * monitor performance of the service is within defined SLO
* I need to discover metrics exposed by the services I subscribed through my preferred platform, 
* I need my app to access these metrics through the platform

As a service user, the metrics discovery is quite similar to service discovery in the catalog (along with discovery of service params, binding response, etc...). I.e. I can discover the list of metrics exposed to me, their format and a potential human description describing their semantics.

As a broker author, in order to provide metrics to service users and bound apps, I need a standard way of exposing metrics to platforms:
* a secured, authenticated way to provide metrics associated to each service instance and possibly service binding
* a portable way to provide metrics across platforms:
   * available to both public services and private on-prem services (i.e. the service broker is'nt required to acquire privileged platform permissions to expose metrics to end-users)

Related pointers:
* [F2F Nov 2017 Topic: Metrics, health, logging - Problem Discovery](https://docs.google.com/document/d/1CNQzUix8snLJJ8QnK6T_XIx5xoCkH9sDFwisV-8Rekg/edit#)
* CF [Loggregator Service Drains](https://docs.google.com/document/d/1TWNucsNDbQD9Oxmc4ibOsKPdgmcw-AAPM-qAfnABdNU/edit?disco=AAAABc6GKXQ)
   * [cloudfoundry/service-metrics](https://github.com/cloudfoundry/service-metrics) and its associated [service-metrics-release](https://github.com/cloudfoundry/service-metrics-release)

**Comment by pmorie** [comment]
> I think this is a case where the ask is out of scope for OSBAPI but the hooks are getting in place for a broker to provide this functionality through a combination of Generic Actions and Broker Generic Actions.

I could see a metrics scrap endpoint being associated with the instance via generic actions.

**Comment by n3wscott** [comment]
OK I am back, give me a little bit of time to work on this now. My Q2 is a little tight on time so feel free for someone else to pick up this task or work with me.

**Comment by mattmcneeney** [comment]
I agree that the Generic Actions proposal #431 is going to be the right place to solve this. What we will want (somewhere) is an example (or standard?) OpenAPI doc for a service instance that wants to expose a metrics endpoint. Maybe this issue can be used to track the creation of that standard?

**Comment by gberche-orange** [comment]
For the CF community, the [related epic](https://www.pivotaltracker.com/epic/show/4316905) in SAPI backlog, if this may help follow CF exchanges on the topic.

**Comment by mattmcneeney** [comment]
Closing this. Work on the generic extensions work can be tracked in #670 !

**Comment by mattmcneeney** [comment]
Reopened, updated title and added the blocked label so that we tackle this after the generic extensions (#670) work is done.
